apps/root/src/pages/token-profile/historical-prices/index.tsx

```diff
@@ -0,0 +1,356 @@
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                {type === TransactionEventTypes.SWAP ? (
+                  <FormattedMessage
+                    description="token-profile-historical-prices.tooltip.swapped"
+                    defaultMessage="Swapped:"
+                  />
+                ) : (
+                  <FormattedMessage
+                    description="token-profile-historical-prices.tooltip.transfered"
+                    defaultMessage="Transfered:"
+                  />
+                )}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                <Address address={user} trimAddress />
+              </Typography>
+            </ContainerBox>
+            <ContainerBox alignItems="center" gap={1}>
+              <TokenIcon token={value.token} size={5} />
+              <Typography variant="bodySmallBold" color={({ palette: { mode } }) => colors[mode].typography.typo2}>
+                {formatCurrencyAmount({ amount: value.amount.amount, token: value.token, intl })}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                (${formatUsdAmount({ intl, amount: value.amount.amountInUSD })})
+              </Typography>
+            </ContainerBox>
+          </ContainerBox>
+        ))
+      )}
+    </StyledTooltipContainer>
+  );
+};
+
+const PeriodMap: Record<Exclude<GraphContainerPeriods, GraphContainerPeriods.all>, TimeString> = {
+  [GraphContainerPeriods.day]: '30m',
+  [GraphContainerPeriods.week]: '3h',
+  [GraphContainerPeriods.month]: '12h',
+  [GraphContainerPeriods.year]: '1w',
+};
+
+const DATA_POINTS = 200;
+
+const TokenHistoricalPrices = ({ token }: TokenHistoricalPricesProps) => {
+  const mode = useThemeMode();
+  const priceService = usePriceService();
+  const providerService = useProviderService();
+  const [selectedPeriod, setSelectedPeriod] = React.useState<GraphContainerPeriods>(GraphContainerPeriods.month);
+  const prevPeriod = usePrevious(selectedPeriod);
+  const [isLoadingHistorical, setIsLoadingHistorical] = React.useState(true);
+  const [graphPrices, setGraphPrices] = React.useState<Partial<Record<GraphContainerPeriods, PriceResult[]>>>({});
+
+  const handlePeriodUpdate = (newPeriod: GraphContainerPeriods) => {
+    if (!graphPrices[newPeriod]) {
+      // When period changes, the useMemo renders no data before the isLoading is set to true by the useEffect
+      // By setting it here, we avoid the flickering of the graph
+      setIsLoadingHistorical(true);
+    }
+
+    setSelectedPeriod(newPeriod);
+  };
+
+  React.useEffect(() => {
+    const handleAllTimeframe = async () => {
+      const firstTimestamp = await providerService.getBlockTimestamp(token.chainId, 1n);
```

Comment:
We can later explore this endpoint to improve this timestamp lookup:
<img width="616" alt="Screenshot 2024-08-05 at 11 18 39 AM" src="https://github.com/user-attachments/assets/ef5ff202-c2c9-40ef-b571-a3411f92d8ac">